app/src/test/java/org/mozilla/fenix/share/ShareControllerTest.kt

```diff
@@ -108,6 +108,8 @@ class ShareControllerTest {
         testController.handleShareToApp(appShareOption)
         advanceUntilIdle()
 
+        assertEquals(appPackageName, Events.shareToApp.testGetValue()?.last()?.extra?.getValue("app_package"))
```

Comment:
Let's also check the assert the size https://searchfox.org/mozilla-mobile/source/fenix/app/src/test/java/org/mozilla/fenix/components/metrics/MetricControllerTest.kt#373

Reply:
+1 on the gl comment, it appears to be a common pattern (which I myself just learnt about)

Reply:
To expand a bit further, we want to add the null assertion prior to make sure we didn't already have a value prior to recording the telemetry ping and also to check that we aren't sending recording more values than expected.